spring/spring-boot-3/remote/src/main/java/org/infinispan/spring/starter/remote/InfinispanRemoteCacheManagerAutoConfiguration.java

```diff
@@ -19,10 +21,25 @@
 @ConditionalOnProperty(value = "infinispan.remote.cache.enabled", havingValue = "true", matchIfMissing = true)
 public class InfinispanRemoteCacheManagerAutoConfiguration {
 
+   @Autowired
+   private InfinispanRemoteConfigurationProperties infinispanProperties;
+
    @Bean
    @ConditionalOnBean(RemoteCacheManager.class)
    @ConditionalOnMissingBean(type = {"org.infinispan.spring.remote.provider.SpringRemoteCacheManager", "org.infinispan.spring.remote.provider.SpringRemoteCacheManagerFactoryBean"})
    public SpringRemoteCacheManager springRemoteCacheManager(RemoteCacheManager remoteCacheManager) {
-      return new SpringRemoteCacheManager(remoteCacheManager);
+      long readTimeout =  infinispanProperties.getReadTimeout();
+      long writeTimeout = infinispanProperties.getWriteTimeout();
+      boolean reactive = infinispanProperties.isReactive() == null ? hasReactor() : infinispanProperties.isReactive();
+      return new SpringRemoteCacheManager(remoteCacheManager, reactive, readTimeout, writeTimeout);
+   }
+
+   public boolean hasReactor() {
```

Comment:
@wburns do you think this is ok ?

Reply:
I can't really comment tbh. From a correctness standpoint this does not seem like a good idea. Essentially you have made any write operation asynchronous with no way to be notified when it is complete if reactor is available.

What is the actual issue this is solving? Is there no SPI we can implement to allow for user to use this in a reactive way?

Reply:
No, this is the problem right now. The SPI for put methods consist in changing the javadocs and making put methods fire and forget

Reply:
But I've been thinking about this and maybe what we need is a reactive dependency to be included with reactor instead of this one